app/models/ext_management_system.rb

```diff
@@ -37,7 +37,7 @@ def self.supported_types_for_create
   end
 
   def self.supported_for_create?
-    !reflections.include?("parent_manager")
+    !reflections.include?("parent_manager") && Vmdb::PermissionStores.instance.supported_ems_type?(self.ems_type)
```

Comment:
I'd rather we leverage the existing method that does this (was trying to avoid duplicating this method call).  So, perhaps something like...

```ruby
def self.supported_for_create?
  supported_types_and_descriptions_hash.key?(ems_type) && !reflections.include?("parent_manager")
end
```

I know that method might feel a bit expensive, but it seems better to latch onto it, since that's where the check is already happening.

Alternately, we may just want to elevate the check into the `support_subclasses` method directly, and then the check becomes

```ruby
supported_subclasses.include?(self) && !reflections.include?("parent_manager")
```

Reply:
Forgot to mention that we would need to check the current callers of supported_subclasses to see if we can elevate it.  I'm not sure why the permission check wasn't just put there in the first place.